plugins/obs-qsv11/QSV_Encoder_Internal.cpp

```diff
@@ -143,13 +154,13 @@ mfxStatus QSV_Encoder_Internal::Open(qsv_param_t *pParams, enum qsv_codec codec)
 #if defined(_WIN32)
 	if (m_bUseD3D11)
 		// Use D3D11 surface
-		sts = Initialize(m_ver, &m_session, &m_mfxAllocator,
+		sts = Initialize(&m_ver, &m_session, &m_mfxAllocator,
```

Comment:
We need to pass through the version all the way from initialization. By taking a reference here it means the pointer didnt get propagated correctly. And when starting an encoding session I see unitialized version data in the log

```
info:       major:          1
            minor:          0
```

Which breaks this check https://github.com/obsproject/obs-studio/blob/07a66fb226700da8fa14da8c6099a30ea2353312/plugins/obs-qsv11/obs-qsv11.c#L937

Reply:
It looks like this still the case with the latest commits.

Reply:
Just added new commit. Added a getVersion function to QSV_Encoder_Internal to propagate the version pointer correctly.

Reply:
This leaves code with misleading and unused function behind, such as passing down the `&version` out parameter to `QSV_Encoder_Internal` but instead using `QSV_Encoder_Internal::getVersion` to extract the version.

Its a bit more work to actually cleanup this usage of out parameters, and will clutter the original commit you authored. So if you want to do that its best to do it as a separate commit in this PR, splitting the changes for this cleanup and the changes related to dgpu/adapter selection.

I think its simpler to leave the out parameter usage for a later cleanup and just make the member a pointer type, and leave a larger cleanup to later.

Reply:
@thyintel As far as I can tell, this feedback still requires action. Do you have any questions about this feedback?

Reply:
So sorry for the wait. This issue should be resolved. We added back the MFXQueryVersion() back to when QSV_Encoder_Internal is initialized. We will look to clean this up later, as we should fix the adapter reordering first.

Reply:
Please also revert changes to `Initialize` so it takes a `mfxVersion ver` since we dont need it with the `MFXQueryVersion` added back.

Reply:
Changed it back. I may not need the adapter_index variable anymore now, but let me know what you think. This should be the only change left, if needed?

Reply:
It should be good now, if you have a log from this commit I can take a quick look or you can wait for me to test on an intel device in a bit.